elftools/elf/elffile.py

```diff
@@ -211,18 +211,194 @@ def get_dwarf_info(self, relocate_dwarf_sections=True):
+            'EM_MANIK'         : 'M2000 Reconfigurable RISC',
+            'EM_CRAYNV2'       : 'Cray Inc. NV2',
+            'EM_RX'            : 'Renesas RX',
+            'EM_METAG'         : 'Imagination Technologies META',
+            'EM_MCST_ELBRUS'   : 'MCST Elbrus',
+            'EM_ECOG16'        : 'Cyan Technology eCOG16',
+            'EM_CR16'          : 'National Semiconductor CompactRISC CR16 16-bit',
+            'EM_ETPU'          : 'Freescale',
+            'EM_SLE9X'         : 'Infineon Technologies SLE9X',
+            'EM_L10M'          : 'Intel L10M',
+            'EM_K10M'          : 'Intel K10M',
+            'EM_AARCH64'       : 'AArch64',
+            'EM_AVR32'         : 'Atmel 32-bit',
+            'EM_STM8'          : 'STMicroeletronics STM8 8-bit',
+            'EM_TILE64'        : 'Tilera TILE64',
+            'EM_TILEPRO'       : 'Tilera TILEPro',
+            'EM_MICROBLAZE'    : 'Xilinx MicroBlaze 32-bit RISC',
+            'EM_CUDA'          : 'NVIDIA CUDA',
+            'EM_TILEGX'        : 'Tilera TILE-Gx',
+            'EM_CLOUDSHIELD'   : 'CloudShield',
+            'EM_COREA_1ST'     : 'KIPO-KAIST Core-A 1st generation',
+            'EM_COREA_2ND'     : 'KIPO-KAIST Core-A 2nd generation',
+            'EM_ARC_COMPACT2'  : 'Synopsys ARCompact V2',
+            'EM_OPEN8'         : 'Open8 8-bit RISC',
+            'EM_RL78'          : 'Renesas RL78',
+            'EM_VIDEOCORE5'    : 'Broadcom VideoCore V',
+            'EM_78KOR'         : 'Renesas 78KOR',
+            'EM_56800EX'       : 'Freescale 56800EX',
+            'EM_BA1'           : 'Beyond BA1',
+            'EM_BA2'           : 'Beyond BA2',
+            'EM_XCORE'         : 'XMOS xCORE',
+            'EM_MCHP_PIC'      : 'Microchip 8-bit PIC',
+            'EM_INTEL205'      : 'Reserved by Intel',
+            'EM_INTEL206'      : 'Reserved by Intel',
+            'EM_INTEL207'      : 'Reserved by Intel',
+            'EM_INTEL208'      : 'Reserved by Intel',
+            'EM_INTEL209'      : 'Reserved by Intel',
+            'EM_KM32'          : 'KM211 KM32 32-bit',
+            'EM_KMX32'         : 'KM211 KMX32 32-bit',
+            'EM_KMX16'         : 'KM211 KMX16 16-bit',
+            'EM_KMX8'          : 'KM211 KMX8 8-bit',
+            'EM_KVARC'         : 'KM211 KVARC',
+            'EM_CDP'           : 'Paneve CDP',
+            'EM_COGE'          : 'Cognitive',
+            'EM_COOL'          : 'Bluechip Systems CoolEngine',
+            'EM_NORC'          : 'Nanoradio Optimized RISC',
+            'EM_CSR_KALIMBA'   : 'CSR Kalimba',
+            'EM_Z80'           : 'Zilog Z80',
+            'EM_VISIUM'        : 'VISIUMcore',
+            'EM_FT32'          : 'FTDI Chip FT32 32-bit RISC',
+            'EM_MOXIE'         : 'Moxie',
+            'EM_AMDGPU'        : 'AMD GPU',
+            'EM_RISCV'         : 'RISC-V'
+        }
+
+        if self['e_machine'] in architectures:
+            return architectures[self['e_machine']]
```

Comment:
It could be a bit cleaner to use the dict .get method with a default of '<unknown>' here?

Reply:
Thanks! My python is a bit rusty.